include/lbann/layers/math/matmul.hpp

```diff
@@ -50,29 +50,43 @@ class matmul_layer : public data_type_layer<TensorDataType> {
 
 public:
 
-  matmul_layer(lbann_comm *comm);
+  matmul_layer(lbann_comm *comm,
+               bool transpose_a = false,
+               bool transpose_b = false);
   matmul_layer(const matmul_layer& other) = default;
   matmul_layer& operator=(const matmul_layer& other) = default;
   matmul_layer* copy() const override;
+
   std::string get_type() const override;
   data_layout get_data_layout() const override;
   El::Device get_device_allocation() const override;
 
+  description get_description() const override;
+
 protected:
 
   void setup_dims() override;
   void fp_compute() override;
   void bp_compute() override;
 
+private:
+
+  /** Whether to transpose matrices from the first input tensor. */
```

Comment:
Prefer TF-like documentation (e.g., _transpose_a: If True, a is transposed before multiplication_), this is clear enough though